File: sample_recipe/banciyuan.py
# !/usr/bin/env python
# -*- coding:utf-8 -*-

from rank_base import Feed

class Banciyuan(Feed):
    name = 'banciyuan'
    url = r'http://bcy.net/illust/toppost100?type=lastday'
    capture = {}
    capture["catch"] = [".detail_std"]

    def spider_get_post_href(self, soup):
        for e in soup.select(".work-thumbnail__topBd > a"): yield e['href']
    def spider_process_html(self, soup):
        dom = soup.select('img')
        if len(dom) == 0: return None
        for e in dom: return '<img src="%s" />' % e['src'].replace("/w650", '')

recipe = Banciyuan

File: sample/banciyuan.js

const Base = require('../common/recipe/rank.js')

module.exports = class Feed extends Base{
  init(){
    super.init()
    this.name = '半次元日榜'
    this.url = 'http://bcy.net/illust/toppost100?type=lastday'
    this.rankCapture.size = 10
    this.rankCapture.catch = ['.js-workTopList ._box>a']
    this.capture.catch = ['.post__title','.w650']
  }
  processImg($, e, url){
    let src = $(e).attr('src')
    if (src) $(e).attr('src', src.replace("/w650", ''))
    return super.processImg($, e, url)
  }
}
